ibis/expr/types/joins.py

```diff
@@ -168,7 +168,7 @@ def prepare_predicates(
 
     left, right = chain.to_expr(), right.to_expr()
     for pred in util.promote_list(predicates):
-        if isinstance(pred, (Value, Deferred, bool)):
+        if isinstance(pred, (Value, bool)):
```

Comment:
I'm a little terrified that simply deleting this didn't break any existing tests. Do you know what this is for? If it's on purpose, we should add a test.

Reply:
It was used for the following case: `_.left_field == t.right_field` 

The problem is that a predicate possibly references two tables but the deferred is a single object, we cannot easily (and always correctly) decompose it into two sides. Hence we require the deferred predicates to use the left table as the placeholder.

Reply:
IIRC there was a failing test case at the time of the refactoring. If that's not the case then we should add one to clarify this use case.

Reply:
I wrote a more general overview of this behavior [here](https://github.com/ibis-project/ibis/issues/10703#issuecomment-2613725392), let's move discussion there, since it is more about the spec, and not the implementation.